libtock/console.c

```diff
@@ -81,27 +80,41 @@ int putnstr_async(const char *str, size_t len, subscribe_cb cb, void* userdata)
   void* buf = (void*) str;
 #pragma GCC diagnostic pop
 
-  ret = allow_readonly(DRIVER_NUM_CONSOLE, 1, buf, len);
-  if (ret < 0) return ret;
+  allow_ro_return_t ro = allow_readonly(DRIVER_NUM_CONSOLE, 1, buf, len);
+  if (ro.success == 0) {
+    return tock_error_to_rcode(ro.error);
+  }
 
-  ret = subscribe(DRIVER_NUM_CONSOLE, 1, cb, userdata);
-  if (ret < 0) return ret;
+  subscribe_return_t sub = subscribe2(DRIVER_NUM_CONSOLE, 1, cb, userdata);
+  if (sub.success == 0) {
+    return tock_error_to_rcode(sub.error);
+  }
 
-  ret = command(DRIVER_NUM_CONSOLE, 1, len, 0);
-  return ret;
+  syscall_return_t com = command2(DRIVER_NUM_CONSOLE, 1, len, 0);
+  if (com.type >= TOCK_SYSCALL_SUCCESS) {
+    return TOCK_SUCCESS;
```

Comment:
This also applies in several other places

Reply:
What if its a SUCCESS_U32?

Reply:
then there is a bug in the console driver, because Tock 2.0 states that a given `command_num` can only return a single success type? If it was a `SUCCESS_U32`, then we would be silently dropping the `u32` here anyway.

Reply:
Ah! Right. Sorry, this was a bad bit of code replication.

Reply:
Replaced with both a check and generating an error if it's not the the right one, because if not grabbing data[0] might not be a valid ErrorCode.